addons/purchase/partner.py

```diff
@@ -28,23 +30,30 @@ class res_partner(osv.osv):
     def _purchase_invoice_count(self, cr, uid, ids, field_name, arg, context=None):
         PurchaseOrder = self.pool['purchase.order']
         Invoice = self.pool['account.invoice']
-        return {
-            partner_id: {
-                'purchase_order_count': PurchaseOrder.search_count(cr,uid, [('partner_id', 'child_of', partner_id)], context=context),
-                'supplier_invoice_count': Invoice.search_count(cr,uid, [('partner_id', 'child_of', partner_id), ('type','=','in_invoice')], context=context)
-            }
-            for partner_id in ids
-        }
+        results = {}
+        for partner_id in ids:
+            results[partner_id] = {}
+            try:
+                if 'contracts_count' in field_name:
+                    results[partner_id]['contracts_count'] = AnalyticAccount.search_count(cr, uid, [('partner_id', '=', partner_id)], context=context)
+            except (exceptions.AccessError,):
+                results[partner_id]['journal_item_count'] =  None
+            try:
+                if 'journal_item_count' in field_name:
+                    results[partner_id]['journal_item_count'] = MoveLine.search_count(cr, uid, [('partner_id', '=', partner_id)], context=context)
+            except (exceptions.AccessError,):
+                results[partner_id]['contracts_count'] = None
+        return results
 
     def _commercial_fields(self, cr, uid, context=None):
         return super(res_partner, self)._commercial_fields(cr, uid, context=context) + ['property_product_pricelist_purchase']
 
     _columns = {
         'property_product_pricelist_purchase': fields.property(
-          type='many2one', 
-          relation='product.pricelist', 
+          type='many2one',
```

Comment:
to keep the diff minimal, avoid changing trailing spaces, it makes it harder to read the diff of the commit